src/components/Pressable/GenericPressable/BaseGenericPressable.js

```diff
@@ -0,0 +1,158 @@
+import React, {
+    useCallback,
+    useEffect,
+    useMemo,
+    forwardRef,
+} from 'react';
+import {Pressable} from 'react-native';
+import _ from 'underscore';
+import Accessibility from '../../../libs/Accessibility';
+import HapticFeedback from '../../../libs/HapticFeedback';
+import KeyboardShortcut from '../../../libs/KeyboardShortcut';
+import styles from '../../../styles/styles';
+import genericPressablePropTypes from './PropTypes';
+import CONST from '../../../CONST';
+import * as StyleUtils from '../../../styles/StyleUtils';
+
+/**
+ * Returns the cursor style based on the state of Pressable
+ * @param {Boolean} isDisabled
+ * @param {Boolean} isText
+ * @returns {Object}
+ */
+const getCursorStyle = (isDisabled, isText) => {
+    if (isDisabled) {
+        return styles.cursorDisabled;
+    }
+
+    if (isText) {
+        return styles.cursorText;
+    }
+
+    return styles.cursorPointer;
+};
+
+const GenericPressable = forwardRef((props, ref) => {
+    // eslint-disable-next-line react/destructuring-assignment
+    const {
+        children,
+        onPress,
+        onLongPress,
+        onKeyPress,
+        disabled,
+        style,
+        accessibilityHint,
+        shouldUseHapticsOnLongPress,
+        shouldUseHapticsOnPress,
+        nextFocusRef,
+        keyboardShortcut,
+        shouldUseAutoHitSlop,
+        enableInScreenReaderStates,
+        onPressIn,
+        onPressOut,
+        ...rest
+    } = props;
+
+    const isScreenReaderActive = Accessibility.useScreenReaderStatus();
+    const [hitSlop, onLayout] = Accessibility.useAutoHitSlop();
+
+    const isDisabled = useMemo(() => {
+        let shouldBeDisabledByScreenReader = false;
+        if (enableInScreenReaderStates === CONST.SCREEN_READER_STATES.ACTIVE) {
+            shouldBeDisabledByScreenReader = !isScreenReaderActive;
+        }
+
+        if (enableInScreenReaderStates === CONST.SCREEN_READER_STATES.DISABLED) {
+            shouldBeDisabledByScreenReader = isScreenReaderActive;
+        }
+
+        return props.disabled || shouldBeDisabledByScreenReader;
+    }, [isScreenReaderActive, enableInScreenReaderStates, props.disabled]);
+
+    const onLongPressHandler = useCallback(() => {
+        if (!onLongPress) {
+            return;
+        }
+        if (shouldUseHapticsOnLongPress) {
+            HapticFeedback.longPress();
+        }
+        if (ref.current) {
+            ref.current.blur();
+        }
+
+        onLongPress();
```

Comment:
NAB - Maybe move this up one line (and put the space below it) to match with the below `onPressHandler`